scripts/upgrade_s3_permissions.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+"""
+Requires boto3 and click packages to be installed.
+
+1) Create a new virtualenv (python3 -m venv my_venv).
+2) Enable the virtualenv (my_venv/bin/activate).
+3) Install the dependencies: (pip install boto3 click).
+4) Run the command (--help will give you more information).
+
+What does this script do?
+
+This script ensures that every user (or a named user) has the expected
+permissions for working with versioned S3 buckets.
+"""
+import os
+import click
+import logging
+import sys
+import boto3
+from botocore.exceptions import ClientError
+import json
+from pathlib import Path
+
+
+READWRITE_ACTION = [
+    "s3:GetObject",
+    "s3:GetObjectAcl",
+    "s3:GetObjectVersion",
+    "s3:DeleteObject",
+    "s3:DeleteObjectVersion",
+    "s3:GetObjectVersionAcl",
+    "s3:PutObject",
+    "s3:PutObjectAcl",
+    "s3:RestoreObject",
+    "s3:GetObjectVersionTagging"
+]
+
+LIST_ACTION = [
+    "s3:ListBucket",
+    "s3:GetBucketPublicAccessBlock",
+    "s3:GetBucketPolicyStatus",
+    "s3:GetBucketTagging",
+    "s3:GetBucketPolicy",
+    "s3:GetBucketAcl",
+    "s3:GetBucketCORS",
+    "s3:GetBucketVersioning",
+    "s3:GetBucketLocation",
+    "s3:ListBucketVersions"
+]
+
+LISTUSERBUCKETS_SECTION = {
+    "Sid": "ListUserBuckets",
```

Comment:
Again, the `Sid` here seemed to have changed from `console` to `ListUserBuckets`. Maybe it's more accurate, but it's a risk to introduce this change.

Reply:
IMHO, I think this is quite a small risk here. One positive from this change, is we can see if users have been upgraded

Reply:
But why risk it for a biscuit? The old `console` Sid was quite meaningful as it's clear this statement is there to allow AWS Console access. 

Plus, as I said in another comment, it seems like this `Sid` is actually used programmatically, e.g. at least in here on line `140`, so change it makes this script less idempotent  - reason being once this `Sid` is updated if you want to re-run the script to update these permissions (e.g. because we realise we need another permission) it would not update this statement anymore.

I strongly advice against changing this `Sid` as at best is a cosmetic change at worse can cause real problems so the risk/reward doesn't look great - IMHO.

Reply:
Meaningful, but wrong. For what it's worth the *only* place the Sid is used is in this script.